src/orion/benchmark/task/task.py

```diff
@@ -0,0 +1,298 @@
+        self.seed = seed
+        self.rng = np.random.default_rng(seed)
+        self.hparams.rng = self.rng
+
+    def sample(self, n: int = 1) -> List[HParams]:
+        """Samples `n` random hyperparameter examples from the priors.
+
+        If any dimensions are 'fixed' (using `self.fixed_values`), the returned
+        samples have the corresponding dimensions set to the value in
+        `self.fixed_values`.
+
+        Args:
+            n (int, optional): Number of samples. Defaults to 1.
+
+        Returns:
+            List[HyperParameters]: List of 'random' `HyperParameters` objects.
+        """
+        self.hparams.rng = self.rng
+        samples = [self.hparams.sample() for i in range(n)]
+        for item in samples:
+            # If there are 'fixed' values in `self.fixed_values`, set them on the
+            # sampled objects.
+            for k, v in self.fixed_values.items():
+                if hasattr(item, k):
+                    setattr(item, k, v)
+        return samples
+
+    def sample_np(self, n: int) -> np.ndarray:
+        hps: List[HyperParameters] = self.sample(n)
+        x = np.asarray([hp.to_array() for hp in hps], dtype=np.float32)
+        return x
+
+    # @property
+    # def hparams(self) -> Type[HyperParameters]:
+    #     # NOTE: This might be a bit slow (because of the imports), so it's
+    #     # better to use a concrete subclass of `ProfetTask` like SvmTask or
+    #     # FcNetTask, etc.
+    #     if self.benchmark == "svm":
+    #         from . import SvmTaskHParams
+    #         return SvmTaskHParams
+    #     elif self.benchmark == "forrester":
+    #         from . import ForresterTaskHParams
+    #         return ForresterTaskHParams
+    #     elif self.benchmark == "fcnet":
+    #         from . import FcNetTaskHParams
+    #         return FcNetTaskHParams
+    #     elif self.benchmark == "xgboost":
+    #         from . import XgBoostTaskHParams
+    #         return XgBoostTaskHParams
+    #     else:
+    #         raise RuntimeError(
+    #             f"Benchmark {self.benchmark} doesn't have a registered hparams class! :(")
+
+    def make_dataset(self, n_samples: int) -> TensorDataset:
+        """ Create a Dataset containing hps and performances from `task`. """
+
+        x, y = self.make_dataset_np(n_samples)
+        x = torch.as_tensor(x, dtype=torch.float)
+        y = torch.as_tensor(y, dtype=torch.float)
+        dataset = TensorDataset(x, y)
+        return dataset
+
+    def make_dataset_np(self, n_samples: int) -> Tuple[np.ndarray, np.ndarray]:
+        """ Create a Dataset containing hps and performances from `task`. """
+        hps = self.sample(n_samples)
+        perfs = self(hps)
+        x = np.asarray([hp.to_array() for hp in hps], dtype=np.float32)
+        y = np.asarray(perfs, dtype=np.float32)
+        y = y.reshape([-1])
+        return x, y
+
+    def to_hp_dict(self, hp: Union[HParams, Dict, np.ndarray], **kwargs) -> Dict:
```

Comment:
Why do we need to supports these different formats? The task should be always called the same way in the benchmark.

Reply:
True! Removed this.